### PR TITLE
Improving visual QA + System-initiated notifications

### PR DESCRIPTION
**Pull request by ovenmitt**
-     Adding MRTK voice commands for simple system commands (next, previous step)
-     Adding visual and task context to QA
-     Overall UX improvements for the QA
-     Merging angel_system with the latest version of the unity ARUI plugin
-     ARUI shows notification and provides audio feedback when system detects a skip error

### CODE REVIEW

**Comment by Purg** [comment]
@ovenmitt For any binary files included here, do you think we could add those as git-LFS tracked items? I know there are some complexities around quotas and bandwidth limitations, so I'm not sure if you are aware of those for public github projects. It is ultimately the "right thing to do" with git to keep binaries files out of the git history.

**Comment by ovenmitt** [comment]
> @ovenmitt For any binary files included here, do you think we could add those as git-LFS tracked items? I know there are some complexities around quotas and bandwidth limitations, so I'm not sure if you are aware of those for public github projects. It is ultimately the "right thing to do" with git to keep binaries files out of the git history. If we go this route, we will have to filter through commits to replace file contributions with GitLFS usage, or we could possible squash all of these changes into a single commit for simplification.

I will remove the files, since they are not necessary for the ARUI to work at this point.

**Comment by Purg** [comment]
Hi @ovenmitt! I have a rebase of this branch onto latest master as well as a fix for the code style of the files causing that CI failure. Do you mind if I force-push that to here?

File: ros/angel_system_nodes/angel_system_nodes/audio/question_answerer.py [hunk 1]
```diff
@@ -130,15 +189,45 @@ def publish_generated_response(
...
+                        "content": [
+                            { 
+                                "type": "text",
+                                "text": "Use the image to answer the question."
+                                + prompt,
+                            },
+                            {
+                                "type": "image_url",
+                                "image_url": {
+                                    "url": "data:image/jpeg;base64,"+self.image_msg
```
Comment (derkmed) [inline ros/angel_system_nodes/angel_system_nodes/audio/question_answerer.py:220]: Oh this is soooo exciting!

**Comment by Purg** [comment]
I have a fix for what is causing the code-style CI to fail, but I can't seem to push it here, despite the `Maintainers are allowed to edit this pull request.` seeming to be enabled. The following diff patch file may be applied with `git apply style-fix.txt`

[style-fix.txt](https://github.com/user-attachments/files/15535638/style-fix.txt)

File: ros/angel_system_nodes/angel_system_nodes/task_monitoring/global_step_predictor.py [hunk 2]
```diff
@@ -29,7 +30,8 @@
 PARAM_CONFIG_FILE = "config_file"
 PARAM_ACTIVITY_CONFIG_FILE = "activity_config_file"
 PARAM_TASK_STATE_TOPIC = "task_state_topic"
-PARAM_TASK_ERROR_TOPIC = "task_error_topic"
+PARAM_TASK_ERROR_TOPIC = "system_notification_topic"
```
[the remark this answers is not in the document]
Comment (ovenmitt) [reply ros/angel_system_nodes/angel_system_nodes/task_monitoring/global_step_predictor.py]: Just updated it

**Comment by Purg** [comment]
Can you run `./angel-workspace-shell.sh -- black .` in your terminal, observe the changed and then commit those? That *should* be what is necessary to address the style issues. The items that I keep seeing crop up in this CI job are consistent with what that call fixes.

**Pull request merged**